src/graphs/SMG.cc

```diff
@@ -38,6 +39,16 @@ const std::map<SMGValue, SMGEdgePointsToPtr>& SMG::GetPTEdges() const { return p
 
 const SMGEntitySet<const SMGEdgeHasValue>& SMG::GetHVEdges() const { return hv_edges_; }
 
+const SMGEntitySet<const SMGEdgeHasValue> SMG::GetHVEdges(
+    const SMGEdgeHasValueFilter& filter) const {
+  SMGEntitySet<const SMGEdgeHasValue> to_return;
+  std::copy_if(hv_edges_.begin(),
+               hv_edges_.end(),
+               std::inserter(to_return.set(), to_return.begin()),
+               filter);
+  return to_return;
+}
```

Comment:
FilterSet filters any set of HV edges you give it as argument. This method filters HV edges of particular SMG (you don't have to provide HV edges set explicitly).